rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,206 @@
+* The block validator will reject blocks produced after the stop slot that
+  contain any transaction or any non-zero fee.
+* The node should notify the user at each slot when transaction processing halts
+  in less than *TBD* slots.
+
+To stop the network after a certain slot, the procedure will be as described
+next:
+
+* There will be a configuration parameter set at compile-time that will define
+  the slot at which the node will stop the network.
+* The previous configuration should be overridable at runtime by optional CLI
+  flags.
+* After the configured slot, the block producer will stop producing any blocks.
+* The block validator will reject any blocks received after the stop network
+* slot.
+* The node should notify the user at each slot when block production/validation
+  halts in less than *TBD* slots.
+
+Each of these procedures will be described in detail in the following sections.
+
+### Compile-time configuration and CLI flag
+
+The configuration parameters `slot_tx_end` and `slot_chain_end` will be set at
+compile-time and will define the slot at which the node will stop processing
+transactions and the slot at which the network stops, respectively. These
+configuration parameters will be optional and will default to `None`. If
+`slot_tx_end` is set to `None`, the node will not stop processing transactions.
+If `slot_chain_end` is set to `None`, the node will not stop producing or
+validating blocks.
+
+There will be four optional CLI flags that will override this configuration.
+
+* `--enable-slot-tx-end <slot>` enables the stop transaction processing feature
+  and sets `<slot>` as the slot at which the node will stop processing
+  transactions.
+* `--disable-slot-tx-end` disables the previously mentioned feature and sets the
+  slot at which the node will stop processing transactions to `None`.
+* `--enable-slot-chain-end <slot>` enables the stop network feature and sets
+  `<slot>` as the slot at which the node will stop producing or validating
+  blocks.
+* `--disable-slot-tx-end` disables the previously mentioned feature and sets the
+  slot at which the node will stop producing or validating blocks to `None`.
+
+### Client submits transaction
+
+When a client sends a transaction to the node daemon, the node will check if
+the stop transaction slot configuration is set. If so, and the current global
+slot is less than the configured stop slot, the transaction will be accepted by
+the node and processed as usual. If the current global slot is equal or greater 
+than the configured stop slot, the transaction will be rejected. The client will
+be notified of the rejection and the reason why the transaction was rejected.
+This improves user UX by rejecting transactions that will not be included in the
+ledger in the preceding network.
+
+This can be done by adding these checks and subsequent rejection messages to the
+`mina_commands.ml` functions that handle transactions from the client.
```

Comment:
+1

Reply:
From an [earlier discussion](https://github.com/MinaProtocol/mina/pull/14138#discussion_r1329380807), we got the idea that it would be best not to change the txn pool to reject transactions. We will keep this change to the GraphQL layer then.